core/components/molecules/modal/Modal.tsx

```diff
@@ -217,6 +217,12 @@ class Modal extends React.Component<ModalProps, ModalState> {
       ['Overlay-container--open']: open,
     });
 
+    const bodyClass = classNames({
+      ['Modal-body']: true,
+      ['Modal-body--withMargin']: (headerOptions || footerOptions || footer || header) ? !!footer : true,
+      ['Modal-body--withPadding']: (headerOptions || footerOptions || footer || header) ? !footer : true,
```

Comment:
We can separate this (`(headerOptions || footerOptions || footer || header) ? !!footer : true`) condition in a variable to avoid repeatability.